File: CS/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraEditors.Filtering;
using DevExpress.XtraEditors.Repository;
using DevExpress.Data.Filtering.Helpers;

namespace UnboundFilterControl {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e) {
            CreateFilterColumns();
        }

        private void CreateFilterColumns() {
            filterControl1.FilterColumns.Add(new UnboundFilterColumn("Name", "Field1", typeof(string), new RepositoryItemTextEdit(), FilterColumnClauseClass.String));
            filterControl1.FilterColumns.Add(new UnboundFilterColumn("Age", "Field2", typeof(int), new RepositoryItemSpinEdit(), FilterColumnClauseClass.Generic));
        }

        private void filterControl1_FilterChanged(object sender, DevExpress.XtraEditors.FilterChangedEventArgs e) {
            labelControl1.Text = filterControl1.FilterCriteria.LegacyToString(); // SQL-like filter
        }
    }
}

File: VB/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraEditors.Filtering
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.Data.Filtering.Helpers

Namespace UnboundFilterControl
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			CreateFilterColumns()
		End Sub

		Private Sub CreateFilterColumns()
			filterControl1.FilterColumns.Add(New UnboundFilterColumn("Name", "Field1", GetType(String), New RepositoryItemTextEdit(), FilterColumnClauseClass.String))
			filterControl1.FilterColumns.Add(New UnboundFilterColumn("Age", "Field2", GetType(Integer), New RepositoryItemSpinEdit(), FilterColumnClauseClass.Generic))
		End Sub

		Private Sub filterControl1_FilterChanged(ByVal sender As Object, ByVal e As DevExpress.XtraEditors.FilterChangedEventArgs) Handles filterControl1.FilterChanged
			labelControl1.Text = filterControl1.FilterCriteria.LegacyToString() ' SQL-like filter
		End Sub
	End Class
End Namespace